model/atmosphere/diffusion/src/icon4py/model/atmosphere/diffusion/diffusion.py

```diff
@@ -542,44 +540,31 @@ def _do_diffusion_step(
 
         """
         klevels = self.grid.num_levels
```

Comment:
unrelated to this PR but since we're here: why not call this `num_levels` instead of `klevels`?
```suggestion
        num_levels = self.grid.num_levels
```